[quant][pt2e] Fix a bug in reference quantized module (decomposed mode)

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98903

Summary:
Fixed quant_min/quant_max for per channel quantized weight for reference quantized module in decomposed mode,
this bug is triggered while onboard an internal model

Test Plan:
python test/test_quantization.py TestQuantizeFx.test__convert_to_reference_decomposed_fx_per_channel_quant_module

Reviewers:

Subscribers:

Tasks:

Tags:

jerryzh168 committed:
[quant][pt2e] Fix a but in reference quantized module (decomposed mode)

Summary:
Fixed quant_min/quant_max for per channel quantized weight for reference quantized module in decomposed mode,
this bug is triggered while onboard an internal model

Test Plan:
python test/test_quantization.py TestQuantizeFx.test__convert_to_reference_decomposed_fx_per_channel_quant_module

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

andrewor14 commented on test/quantization/fx/test_quantize_fx.py:
```diff
@@ -6158,6 +6159,56 @@ def forward(self, x, weight, bias):
         res = m(*example_inputs)
         self.assertEqual(res, res_ref)
 
+    def test__convert_to_reference_decomposed_fx_per_channel_quant_module(self):
```
double underscore

jerryzh168 replied:
yeah this is correct since we are testing `_convert_to_reference_decomposed_fx`

jerryzh168 commented:
@pytorchbot rebase

jerryzh168 commented:
@pytorchbot merge -f "test failures unrelated to PR"

jerryzh168 committed:
Update on "[quant][pt2e] Fix a bug in reference quantized module (decomposed mode)"


Summary:
Fixed quant_min/quant_max for per channel quantized weight for reference quantized module in decomposed mode,
this bug is triggered while onboard an internal model

Test Plan:
python test/test_quantization.py TestQuantizeFx.test__convert_to_reference_decomposed_fx_per_channel_quant_module

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

huydhn commented:
@pytorchbot revert -m 'Sorry for reverting your PR, but it breaks Windows tests on trunk https://hud.pytorch.org/pytorch/pytorch/commit/a2e809f29bd66a0f314edeb602f37b4de05e5c41' -c ignoredsignal

jerryzh168 committed:
Update on "[quant][pt2e] Fix a bug in reference quantized module (decomposed mode)"


Summary:
Fixed quant_min/quant_max for per channel quantized weight for reference quantized module in decomposed mode,
this bug is triggered while onboard an internal model

Test Plan:
python test/test_quantization.py TestQuantizeFx.test__convert_to_reference_decomposed_fx_per_channel_quant_module

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][pt2e] Fix a bug in reference quantized module (decomposed mode)"


Summary:
Fixed quant_min/quant_max for per channel quantized weight for reference quantized module in decomposed mode,
this bug is triggered while onboard an internal model

Test Plan:
python test/test_quantization.py TestQuantizeFx.test__convert_to_reference_decomposed_fx_per_channel_quant_module

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge
